runtime/commonMain/src/kotlinx/serialization/json/Json.kt

```diff
@@ -214,14 +187,24 @@ public class JsonBuilder {
     public var encodeDefaults: Boolean = true
     public var strictMode: Boolean = true
     public var unquoted: Boolean = false
+    public var allowStructuredMapKeys: Boolean = false
```

Comment:
I would stick with the verb "enable", we enable features, not allow them

Reply:
I'm not sure that `enableStructuredMapKeys` is OK, because structured map keys exist (are enabled) in the user's data model regardless of JSON settings. `enableTransformationOfStructuredMapKeys` sounds more correct, but it's too long